### DISCUSSION TITLE
Building Rosetta Drone from source

### DISCUSSION THREAD

**Discussion by jason-w-jones** [RosettaDrone/rosettadrone#226] [Q&A]
Hi,

I was wondering if there was a guide that showed how to build and run Rosetta Drone from scratch.

I have been trying with the latest master branch using Android Studio Flamingo 2022.2.1 Patch 2. It seems to build fine but when I try to run it on a Google Pixel 5 connected via USB I get errors about the following class not being found:

- java.lang.NoClassDefFoundError: Failed resolution of: Lorg/bouncycastle/jce/provider/BouncyCastleProvider

I am happy to try this on a newly installed WIndows / Linux PC but a set of instructions would be really useful.

Many thanks,

Jason

**Comment by kripper** [comment]
Hi @jason-w-jones 

Does RD's source code reference this class?

**Reply by jason-w-jones** [reply]
I don't think so but it seems the DJI SDK does. I have included the stack trace when the app is run on the phone via Android Studio. I went down to the last couple of methods in Rosetta Drone.

```
java_vm_ext.cc:591] JNI DETECTED ERROR IN APPLICATION: JNI NewGlobalRef called with pending exception java.lang.NoClassDefFoundError: Failed resolution of: Lorg/bouncycastle/jce/provider/BouncyCastleProvider;
java_vm_ext.cc:591]   at void com.dji.megatronking.stringfog.fdd.fdd.<clinit>() ((null):-1)
java_vm_ext.cc:591]   at void dji.midware.encryption.util.AES256Encryption.<clinit>() ((null):-1)
java_vm_ext.cc:591]   at java.lang.String java.lang.Runtime.nativeLoad(java.lang.String, java.lang.ClassLoader, java.lang.Class) (Runtime.java:-2)
java_vm_ext.cc:591]   at java.lang.String java.lang.Runtime.nativeLoad(java.lang.String, java.lang.ClassLoader) (Runtime.java:1126)
java_vm_ext.cc:591]   at void java.lang.Runtime.loadLibrary0(java.lang.ClassLoader, java.lang.Class, java.lang.String) (Runtime.java:1080)
java_vm_ext.cc:591]   at void java.lang.Runtime.loadLibrary0(java.lang.Class, java.lang.String) (Runtime.java:1003)
java_vm_ext.cc:591]   at void java.lang.System.loadLibrary(java.lang.String) (System.java:1661)
java_vm_ext.cc:591]   at void dji.midware.natives.SDKRelativeJNI.<clinit>() ((null):-1)
java_vm_ext.cc:591]   at void dji.sdk.sdkmanager.DJISDKManager.<clinit>() ((null):-1)
java_vm_ext.cc:591]   at void sq.rogue.rosettadrone.DJISimulatorApplication.onCreate() (DJISimulatorApplication.java:160)
java_vm_ext.cc:591]   at void sq.rogue.rosettadrone.RDApplication.startLoginApplication() (RDApplication.java:42)
```

**Comment by kripper** [comment]
AFAIR, there were some "class not found" messages, but they didn't prevent RD from running normally.

Is RD crashing on you device?

**Comment by kripper** [comment]
Did you try running a simple SDK sample?

**Reply by jason-w-jones** [reply]
Sorry I haven't responded for a while - I have been away for a few days.

To answer your question, RD is crashing - it starts briefly on the phone and then disappears.

Tomorrow, I'll try these things and update you:

- Create an APK and transfer it to the phone so it runs without Android Studio (just to see if it's the USB connection to Studio that is breaking it)
- Try some of the DJI samples with Android Studio connected
- Try debugging RD with Android Studio but over WiFi instead of USB

**Reply by jason-w-jones** [reply]
Am I using rhe correct Android Studio version? I tried the newest version first but there were a ton of Gradle issues so I moved back a few versions to one that used a similar version of Gradle.

**Comment by kripper** [comment]
Please create an issue, not a discussion.
